openghg/analyse/_scenario.py

```diff
@@ -464,6 +464,8 @@ def add_flux(
                 flux_source = self._get_data(flux_keywords, data_type="flux")
                 # TODO: May need to update this check if flux_source is empty FluxData() object
                 if flux_source is not None:
+                    if name is None and len(sources) == 1:
+                        name = flux_source.metadata.get("source", "no_source_specified")
```

Comment:
As discussed on the Issue, if we're unable to extract a "source" value we should be raising an appropriate error as all flux data should have an associated source value and if it doesn't there's a separate issue.

Reply:
I added error handling now.